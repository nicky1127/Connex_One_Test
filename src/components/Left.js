import React, { useEffect, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { Paper, List, ListItem, ListItemIcon, ListItemText } from '@material-ui/core';
import { AccessTime } from '@material-ui/icons';
import client from '../Client';
import Loader from './Loader';

const useStyles = makeStyles(() => ({
  root: {
    width: '100%',
    height: '100%',
    flex: '1 1 auto'
  }
}));
let timer;

const Left = () => {
  const classes = useStyles();
  const [loading, SetLoading] = useState(false);
  const [time, setTime] = useState(Math.floor(Date.now() / 1000));
  const [diffTime, setDiffTime] = useState('00:00:00');

  const getTime = async () => {
    SetLoading(true);
    try {
      const data = await client._get('/time');
      const epoch = data.data.properties.epoch.epochSeconds;
      setTime(epoch);
    } catch (err) {
      console.error('Error getting time: ', err);
    }

    setDiffTime('00:00:00');
    SetLoading(false);
  };
  // eslint-disable-next-line
  const getDiffTime = () => {
    const currentSeconds = Math.floor(Date.now() / 1000);
    const difference = currentSeconds - time;
    const diffSeconds = difference < 10 ? `00:00:0${difference}` : `00:00:${difference}`;
    setDiffTime(diffSeconds);
  };

  useEffect(() => {
    getTime();
    setInterval(() => getTime(), 30000);
  }, []);

  useEffect(() => {
    if (timer) clearInterval(timer);
    timer = setInterval(() => getDiffTime(), 1000);
  }, [time, getDiffTime]);

  const renderList = () => (
    <List>
      <ListItem>
        <ListItemIcon>
          <AccessTime />
        </ListItemIcon>
        <ListItemText primary="Time Retrieved(Epoch): " />
        <ListItemText primary={time} />
      </ListItem>
      <ListItem>
        <ListItemIcon>
          <AccessTime />
        </ListItemIcon>
        <ListItemText primary="Time Difference(sec): " />
        <ListItemText primary={diffTime} />
      </ListItem>
    </List>
  );

  const content = loading ? <Loader /> : renderList();

  return (
    <Paper className={classes.root} elevation={5}>
      {content}
    </Paper>
  );
};

export default Left;
