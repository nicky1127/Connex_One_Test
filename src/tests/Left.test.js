import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Left from '../components/Left';

afterEach(cleanup);

describe('Left', () => {
  it('Snapshot matchs', () => {
    const tree = renderer.create(<Left />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
