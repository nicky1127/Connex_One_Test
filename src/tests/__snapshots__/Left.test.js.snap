// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Left Snapshot matchs 1`] = `
<div
  className="MuiPaper-root makeStyles-root-1 MuiPaper-elevation5 MuiPaper-rounded"
>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
          <path
            d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Time Retrieved(Epoch): 
        </span>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          1596115066
        </span>
      </div>
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
          <path
            d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
          />
        </svg>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          Time Difference(sec): 
        </span>
      </div>
      <div
        className="MuiListItemText-root"
      >
        <span
          className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          00:00:00
        </span>
      </div>
    </li>
  </ul>
</div>
`;
